unity-renderer/Assets/Scripts/MainScripts/DCL/WorldRuntime/BlockerController/WorldBlockersController.cs

```diff
@@ -173,27 +196,30 @@ internal void SetupWorldBlockers(HashSet<Vector2Int> allLoadedParcelCoords)
                     for (int i = 0; i < aroundOffsets.Length; i++)
                     {
                         Vector2Int offset = aroundOffsets[i];
-                        Vector2Int checkedPosition = new Vector2Int(pos.x + offset.x, pos.y + offset.y);
+                        int xCandidate = pos.x + offset.x;
+                        int yCandidate = pos.y + offset.y;
+                        Vector2Int checkedPosition = new Vector2Int(xCandidate, yCandidate);
 
-                        if (!allLoadedParcelCoords.Contains(checkedPosition) && !blockers.ContainsKey(checkedPosition))
+                        if (!dontAddABlockerHere.Contains(checkedPosition) && !blockers.ContainsKey(checkedPosition))
                         {
-                            blockersToAdd.Add(checkedPosition);
+                            blockersCandidate.Add(checkedPosition);
                         }
                     }
                 }
             }
 
-            // Remove extra blockers
-            foreach (var coords in blockersToRemove)
+            if (currentLimitIterationEvaluation == worldBlockersLimit.Get())
+                return blockersCandidate;
+            else
             {
-                blockerInstanceHandler.HideBlocker(coords, false);
+                blockersCandidate.UnionWith(dontAddABlockerHere);
+                return LookForLimits(blockersCandidate, blockers, currentLimitIterationEvaluation + 1);
```

Comment:
I would probably just inline it inside and use ternary operator, but it is very optional (depends on your taste :) )
```
return currentLimitIterationEvaluation == worldBlockersLimit.Get()
                ? blockersCandidate
                : LookForLimits(blockersCandidate.UnionWith(dontAddABlockerHere), blockers, currentLimitIterationEvaluation + 1);

```